src/app/cli/src/commands/system_diagnose_command.rs

```diff
@@ -0,0 +1,246 @@
+    fn needs_workspace(&self) -> bool {
+        false
+    }
+
+    async fn run(&mut self) -> Result<(), CLIError> {
+        let diagnose_check = RunCheck::new(
+            self.dataset_repo.clone(),
+            self.verification_svc.clone(),
+            self.container_runtime.clone(),
+            self.is_in_workpace,
+        );
+        diagnose_check.run().await?;
+        Ok(())
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+pub struct RunCheck {
+    dataset_repo: Arc<dyn DatasetRepository>,
+    verification_svc: Arc<dyn VerificationService>,
+    container_runtime: Arc<ContainerRuntime>,
+    is_in_workpace: bool,
+}
+
+impl RunCheck {
+    fn new(
+        dataset_repo: Arc<dyn DatasetRepository>,
+        verification_svc: Arc<dyn VerificationService>,
+        container_runtime: Arc<ContainerRuntime>,
+        is_in_workpace: bool,
+    ) -> Self {
+        Self {
+            dataset_repo,
+            verification_svc,
+            container_runtime,
+            is_in_workpace,
+        }
+    }
+
+    async fn container_check(&self, f: &mut dyn std::io::Write) -> Result<(), CLIError> {
+        write!(f, "container installed... ")?;
+
+        let container_installed_check_status = match self
+            .container_runtime
+            .custom_cmd("--version".to_string())
+            .output()
+            .await
+        {
+            Ok(_) => style("ok").green(),
+            Err(_) => style("failed").red(),
+        };
+
+        write!(f, "{container_installed_check_status}")?;
+        Ok(())
+    }
+
+    async fn container_pull_check(&self, f: &mut dyn std::io::Write) -> Result<(), CLIError> {
+        write!(f, "container can pull images... ")?;
+
+        let is_container_image_pulled =
+            match self.container_runtime.pull_image(DUMMY_IMAGE, None).await {
+                Ok(_) => style("ok").green(),
+                Err(_) => style("failed").red(),
+            };
+
+        write!(f, "{is_container_image_pulled}")?;
+        Ok(())
+    }
+
+    async fn container_rootless_run_check(
+        &self,
+        f: &mut dyn std::io::Write,
+    ) -> Result<(), CLIError> {
+        write!(f, "container rootless run check... ")?;
+
+        let container_rootless_check_status = match self
+            .container_runtime
+            .run_attached(DUMMY_IMAGE)
+            .init(true)
+            .spawn()
+            .unwrap()
+            .wait()
+            .await
+        {
+            Ok(_) => style("ok").green(),
+            Err(_) => style("failed").red(),
+        };
+
+        write!(f, "{container_rootless_check_status}")?;
+        Ok(())
+    }
+
+    async fn container_volume_mount_check(
+        &self,
+        f: &mut dyn std::io::Write,
+    ) -> Result<(), CLIError> {
+        write!(f, "container volume mounts work... ")?;
+        let temp_dir = tempfile::tempdir()?;
+        let cwd = Path::new(".").canonicalize()?;
```

Comment:
I'm surprised this worked for you. As far as I remember, on MacOS by default only the user home directory is mounted into Docker VM and files mounted from `/tmp` will not be visible to container. This is the reason we create all files for engines in `.kamu/run` directory.

Actually this might be the case with your Mac - you only check that running container succeeds succeeds, but it will even if the path does not exist.

I think we should run container with `cat my-tempfile` command and verify that container could actually read the temporary file.